### PR TITLE
Refactor tpu_client -- remove duplicate code

### PR DESCRIPTION
**Pull request by lijunwangs**
#### Problem
The code of the client/tpu_client and tpu-client/tpu_client code is essentially the same except for the ConnectionCache used. 

#### Summary of Changes
Removed implementation in client/tpu_client  and use  tpu-client/tpu_client by a forwarding declaration. There is some minor change to coerce the client/ConnectionCache to connection-cache/ConnectionCache. But the code should be much streamlined now.

Fixes #

### CODE REVIEW

**Review by CriesofCarrots — APPROVED** [review]
Awesome, thanks! Lgtm

**Comment by lijunwangs** [comment]
@CriesofCarrots downstream-projects is failing due to this. I think we have to leave a thin-wrapper to workaround. Or can we fix the test itself?

**Comment by CriesofCarrots** [comment]
> I think we have to leave a thin-wrapper to workaround. Or can we fix the test itself?

Oh shoot, so sorry, I missed the extra arg in the constructor. I think yes, we might need to leave a thin wrapper, but let me think on it overnight.

**Comment by joncinque** [comment]
Now that 1.15 is cut, we can upgrade SPL to that and use the tpu client directly to avoid the downstream issue. The only issue I can foresee there is during `cargo-install-all.sh` for 1.14, since it would use Rust 1.60 to build the token-cli which targets 1.15. This means we'll have to hold back the token-cli in 1.14.

On the flipside, I have one regression that I need to fix https://github.com/solana-labs/solana-program-library/pull/4017, so once that's done, we can tag a release and then update to 1.15!

**Comment by CriesofCarrots** [comment]
> Now that 1.15 is cut, we can upgrade SPL to that and use the tpu client directly to avoid the downstream issue.

True, and we should do that to simplify spl-token-cli dependencies. However, I think this is an indication that other downstream users will be broken by the change. @lijunwangs I think our best option is that thin wrapper and deprecation message, as we've done with ConnectionCache

File: client/src/tpu_client.rs [hunk 1]
```diff
@@ -1,6 +1,104 @@
...
+        transport::Result as TransportResult,
+    },
+    solana_tpu_client::tpu_client::temporary_pub::Result,
+    std::sync::Arc,
+};
 pub use {
     crate::nonblocking::tpu_client::TpuSenderError,
     solana_tpu_client::tpu_client::{
-        TpuClient, TpuClientConfig, DEFAULT_FANOUT_SLOTS, MAX_FANOUT_SLOTS,
+        TpuClient as BackendTpuClient, TpuClientConfig, DEFAULT_FANOUT_SLOTS, MAX_FANOUT_SLOTS,
```
Comment (CriesofCarrots) [inline client/src/tpu_client.rs:17]: Move `TpuClient as BackendTpuClient` to L11 (the non-pub use statement)

Comment (lijunwangs) [reply client/src/tpu_client.rs]: done

**Pull request merged**